packages/sdk-core/src/model/http/model/BacktraceSubmissionResult.ts

```diff
@@ -0,0 +1,49 @@
+import { BacktraceSubmissionStatus } from './BacktraceSubmissionStatus';
+
+export class BacktraceReportSubmissionResult<T> {
+    public get result(): T | undefined {
+        return this._result;
+    }
+
+    public readonly status: BacktraceSubmissionStatus = 'Ok';
+
+    private _result?: T;
+    private constructor(response: T);
+    private constructor(_status: BacktraceSubmissionStatus, _message?: string);
+    private constructor(statusOrResponse: T | BacktraceSubmissionStatus, public readonly message?: string) {
+        if (this.isSubmissionResponse(statusOrResponse)) {
+            this.status = statusOrResponse;
+            return;
+        }
+        this._result = statusOrResponse;
+    }
+
+    public static OnLimitReached<T>(): BacktraceReportSubmissionResult<T> {
+        return new BacktraceReportSubmissionResult('Limit reached', 'Client report limit reached');
+    }
+
+    public static OnInternalServerError<T>(message: string): BacktraceReportSubmissionResult<T> {
+        return new BacktraceReportSubmissionResult('Server Error', message);
+    }
+
+    public static OnInvalidToken<T>(): BacktraceReportSubmissionResult<T> {
+        return new BacktraceReportSubmissionResult<T>('Invalid token');
+    }
+
+    public static OnUnknownError<T>(message: string): BacktraceReportSubmissionResult<T> {
+        return new BacktraceReportSubmissionResult('Unknown', message);
+    }
+    public static OnNetworkingError<T>(message: string): BacktraceReportSubmissionResult<T> {
+        return new BacktraceReportSubmissionResult('Network Error', message);
```

Comment:
Out of curiosity, why do some of these pass the type `T` to the `BacktraceReportSubmissionResult` and others not?

Reply:
because of the generic function, the function definition should have a generic parameter, because the resulting object is generic. We can apply a default value to it and then the generic parameter will be redundant.